circletree/CircleSegment.jsx

```diff
@@ -75,9 +82,29 @@ var CircleSegment = React.createClass({
     }
   },
 
+  onActivate(e) {
+    this.setState({
+      status: ""
+    });
+    if (e.detail.origin === this) {
+      this.activate("primary");
+    }
+  },
+
+  activate(status) {
+    this.setState({ status });
+    if (this.props.activate) {
+      this.props.activate("active");
+    }
+  },
+
   render() {
+    var className = [
```

Comment:
if you're composing this way, we probably want to require in [`classnames`](https://npmjs.com/package/classnames) and then use that instead